### PR TITLE
feature: SIMD matrices support

### PR DESCRIPTION
**Pull request by tdermendjiev**
Current implementation of libffi doesn't handle ext_vector values properly on Aarch64. New ffi type = FFI_TYPE_EXT_VECTOR is introduced in order for **ffi_call** function to be able to distinguish between standard collection types (marked as FFI_TYPE_STRUCT) and ext_vector ones as ext_vector values are not being stored in general-purpose registers.

### CODE REVIEW

**Comment by Natalia-Hristova** [comment]
run ci

**Comment by ginev** [comment]
run ci

**Comment by ginev** [comment]
It seems the build fails on Xcode 8 machines.

File: src/NativeScript/Marshalling/Reference/ExtVectorTypeInstance.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+const char* ExtVectorTypeInstance::encode(VM& vm, JSCell* cell) {
+    ExtVectorTypeInstance* self = jsCast<ExtVectorTypeInstance*>(cell);
+
+    if (!self->_compilerEncoding.empty()) {
+        return self->_compilerEncoding.c_str();
+    }
+
+    self->_compilerEncoding = "^";
```
Comment (mbektchiev) [inline src/NativeScript/Marshalling/Reference/ExtVectorTypeInstance.cpp]: According to the [Type Encodings documentation article](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtTypeEncodings.html)
>The type code for an array is enclosed within square brackets; the number of elements in the array is specified immediately after the open bracket, before the array type. For example, an array of 12 pointers to floats would be encoded as:
[12^f]

I think that `IndexedRefTypeInstance`'s and `ExtVectorTypeInstance`'s compiler encodings should both be changed accordingly.

**Comment by Natalia-Hristova** [comment]
run ci

**Comment by Natalia-Hristova** [comment]
run ci

**Comment by Natalia-Hristova** [comment]
run ci

**Pull request merged**